lighthouse-core/audits/byte-efficiency/duplicated-javascript.js

```diff
@@ -162,9 +161,8 @@ class DuplicatedJavascript extends ByteEfficiencyAudit {
         /** @type {number|undefined} */
         let transferRatio = transferRatioByUrl.get(url);
         if (transferRatio === undefined) {
-          const networkRecord = url === artifacts.URL.finalUrl ?
-            mainDocumentRecord :
-            networkRecords.find(n => n.url === url);
+          const fakeScript = /** @type {LH.Artifacts.ScriptElement} */ ({src: url});
+          const networkRecord = getRequestForScript(networkRecords, fakeScript);
 
```

Comment:
This is the only place I had to make non-trivial changes when cherry-picking. It is because we lack the (breaking) ScriptElements->Scripts change, yet #13751 built off that.

also ctrl+F fakeScript